lndsigner_test.go

```diff
@@ -0,0 +1,539 @@
+				"passphrase": "weks1234",
+				"node":       "03c7926302ac72f51ef009dc169561734414b3c6bfd9fb0dc42cac93101c3c25bf",
+			},
+			check: func(t *testing.T, resp *api.Secret) {
+				pubKey, ok := resp.Data["node"].(string)
+				require.True(t, ok)
+				require.Equal(t, lnd3PK, pubKey)
+				t.Logf("Imported node %s", pubKey)
+			},
+		},
+	}
+
+	for _, testCase := range initCases {
+		t.Run(testCase.name, func(t *testing.T) {
+			resp, err := client.Write(testCase.path,
+				testCase.reqData)
+			require.NoError(t, err)
+
+			testCase.check(t, resp)
+		})
+	}
+
+	err = os.Setenv("VAULT_TOKEN", "root")
+	require.NoError(t, err)
+
+	err = os.Setenv("VAULT_ADDR", "http://127.0.0.1:"+vaultPort)
+	require.NoError(t, err)
+
+	// Start bitcoind
+	bitcoinDir := path.Join(tmpRoot, "bitcoin")
+	err = os.Mkdir(bitcoinDir, fs.ModeDir|0700)
+	require.NoError(t, err)
+
+	bitcoinRPC := newPortString()
+	bitcoinZB := newPort()
+	bitcoinZT := newPort()
+
+	bitcoindCmd := exec.CommandContext(ctx, bitcoindPath, "-server=1",
+		"-datadir="+bitcoinDir, "-listen=0", "-txindex=1",
+		"-regtest=1", "-rpcuser=user", "-rpcpassword=password",
+		"-rpcport="+bitcoinRPC,
+		"-zmqpubrawblock=tcp://"+bitcoinZB.String(),
+		"-zmqpubrawtx=tcp://"+bitcoinZT.String())
+
+	bitcoindErrPipe, err := bitcoindCmd.StderrPipe()
+	require.NoError(t, err)
+
+	err = bitcoindCmd.Start()
+	require.NoError(t, err)
+	defer killProc(t, bitcoindCmd, bitcoindErrPipe)
+
+	// TODO(aakselrod): eliminate this
+	time.Sleep(time.Second)
+
+	bitcoinCli := func(args ...string) {
+		bitcoinCliCmd := exec.CommandContext(ctx, bitcoincliPath,
+			append([]string{"-datadir=" + bitcoinDir,
+				"-rpcport=" + bitcoinRPC, "-rpcuser=user",
+				"-rpcpassword=password", "-rpcwaittimeout=5"},
+				args...)...)
+
+		stderrPipe, err := bitcoinCliCmd.StderrPipe()
+		require.NoError(t, err)
+
+		err = bitcoinCliCmd.Start()
+		require.NoError(t, err)
+
+		stderr, err := io.ReadAll(stderrPipe)
+		require.NoError(t, err)
+
+		err = bitcoinCliCmd.Wait()
+		require.NoError(t, err, string(stderr))
+	}
+
+	bitcoinCli("createwallet", "default")
+	bitcoinCli("-generate", "1000")
+
+	// Start lnd
+	lnds := []*lndHarness{
+		&lndHarness{
+			lnddir:    path.Join(tmpRoot, "lnd1"),
+			lncliPath: lncliPath,
+			vault:     client,
+		},
+		&lndHarness{
+			lnddir:    path.Join(tmpRoot, "lnd2"),
+			lncliPath: lncliPath,
+			vault:     client,
+		},
+		&lndHarness{
+			lnddir:    path.Join(tmpRoot, "lnd3"),
+			lncliPath: lncliPath,
+			vault:     client,
+		},
+	}
+
+	lndPubKeys := []string{lnd1PK, lnd2PK, lnd3PK}
+
+	var resp map[string]interface{}
+	for i, lnd := range lnds {
+		var address string
+
+		lnd.Start(t, ctx, lndPath, bitcoinRPC, bitcoinZB.String(),
+			bitcoinZT.String(), lndPubKeys[i])
+		defer lnd.Close(t)
+
+		t.Run(lndPubKeys[i]+"/"+"getinfo", func(t *testing.T) {
+			resp = lnd.Lncli(t, ctx, "getinfo")
```

Comment:
Most of the time is taken by the initial setup. Instead of running the tests using `t.Parallel()`, I've manually parallelized the integration tests, including setup, and got a bit of speedup. I can run some parts of setup and some tests in parallel on each node, and some tests in parallel on each pair of nodes.